### ISSUE TITLE
Allow errors and warnings for the same rule with different options

### ISSUE THREAD

**Issue by UziTech** [eslint/eslint#11089]
**The version of ESLint you are using.**

v5.9.0

**The problem you want to solve.**

Allow errors and warnings for the same rule with different options.

The following examples are some use cases that would be beneficial:

 - Warn after line is greater than 80 chars, error after 120 chars.
 - Warn after function is greater than 10 lines, error after 15 lines.
 - Warn for unused function args, error for other unused variables.
 - Warn for console.error/log , error for console.debug.

**Your take on the correct solution to problem.**

Allow `"rules"` configuration to be an object with `"error"`, `"warn"`, and `"off"` properties which are objects with each rule as a property and options for that rule as their value.

e.g.
```js
{
  "rules" : {
    "error": {
      "max-lines": 120,
      "max-lines-per-function": 15,
      "no-unused-vars": {
        "vars": "all",
        "args": "none"
      },
      "no-console": {"allow": ["log", "error"]}
    },
    "warn": {
      "max-lines": 80,
      "max-lines-per-function": 10,
      "no-unused-vars": {
        "vars": "none",
        "args": "after-used"
      },
      "no-console": true
    }
  }
}
```

Rules would be evaluated by *max* severity, so a line that is 130 chars long would match the `warn: max-lines` rule as well as the `error: max lines` rule but only the `error` would be reported since it is a higher severity than `warn`.

`warn: no-console` value is `true` to signify not passing any options, or using the defaults.

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by nzakas** [comment]
Thanks for the proposal. We've had similar proposals in the past and have declined to accept them due to the complexity involved. Not only would this greatly complicate our (already very complicated) configuration system, it seems like a lot of work for what would end up being a very narrow use case that few people would use.

That's just my opinion, though, so let's see what others might think.

**Comment by UziTech** [comment]
I agree it would be a pretty big change, but I believe the benefit would be huge. I would be willing to bet that this feature would eventually be used by everyone and one of those things we didn't know we could live without. Kind of like when the first linter was created.

**Comment by UziTech** [comment]
Not to mention, I think this type of configuration is more intuitive than the current `"rules"` object where the values are arrays where the first value in the array is the severity and the second is the options.

I usually just copy and paste eslint configurations whenever I create a new project but any time I actually need to change something it seems like I have to re-learn the configuration rules.
